### ISSUE TITLE
Gateway Timeout: PHP Issue

### ISSUE THREAD

**Issue by jengajenga** [WWBN/AVideo#6108]
@DanielnetoDotCom hi

when i visit my site i face this error

_"Gateway Timeout: The gateway did not receive a timely response from the upstream server or application."_

The error log says this

[Thu Dec 23 19:08:25.129929 2021] [proxy_fcgi:error] [pid 721096] (70007)The timeout specified has expired: [client 105.160.123.161:0] AH01075: Error dispatching request to : (polling), referer: https://www.xxx.ke/
[Thu Dec 23 19:08:36.717899 2021] [proxy_fcgi:error] [pid 721104] (70007)The timeout specified has expired: [client xxx.xxx.xxx.86:0] AH01075: Error dispatching request to : (polling)
[Thu Dec 23 19:09:06.765911 2021] [proxy_fcgi:error] [pid 721105] (70007)The timeout specified has expired: [client xxx.xxx.xxx.86:0] AH01075: Error dispatching request to : (polling)
[Thu Dec 23 19:09:16.165896 2021] [proxy_fcgi:error] [pid 721106] (70007)The timeout specified has expired: [client 105.160.123.161:0] AH01075: Error dispatching request to : (polling), referer: https://www.xxx.ke/
[Thu Dec 23 19:09:23.736282 2021] [proxy_fcgi:error] [pid 721110] (70007)The timeout specified has expired: [client xxx.xxx.xxx.4:0] AH01075: Error dispatching request to : (polling), referer: https://www.xxx.ke/
[Thu Dec 23 19:09:36.820146 2021] [proxy_fcgi:error] [pid 715488] (70007)The timeout specified has expired: [client xxx.xxx.xxx.86:0] AH01075: Error dispatching request to : (polling)"

I disabled restream because I thought it was part of the problem
To try and fix it, I have increased memory_limit in php from 2GB to 4GB

When I restart php8.0 it continues properly but shortly it starts again with the timeout error. Where should I continue to look?

Thank you for you assistance.

**Comment by DanielnetoDotCom** [comment]
Are you using Apache or Nginx for your Streamer site?

the proxy_fcgi:error is strange for me. looks like you are using a different configuration than mine

**Comment by jengajenga** [comment]
I figured this out. 

PHP Gateway Timeout is when the PHP-FPM doesn't create sufficient servers to handle the incoming requests. The number has to be increased in the php-conf under the pool.d folder in /etc/php

However for this case, I reverted to PHP Apache Handler so Apache handles my PHP. I thought it would be more efficient with PHP-FPM but it caused more problems than I expected. It also made the app not work.

So apache2 handler 2.0 for me going forward. i hope this helps someone else.
